lints/lint_mp_allowed_eku.go

```diff
@@ -0,0 +1,74 @@
+/*
+ * ZLint Copyright 2019 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/********************************************************************
+Section 5.3 - Intermediate Certificates
+Intermediate certificates created after January 1, 2019, with the exception
+of cross-certificates that share a private key with a corresponding root
+certificate: MUST contain an EKU extension; and, MUST NOT include the
+anyExtendedKeyUsage KeyPurposeId; and, * MUST NOT include both the
+id-kp-serverAuth and id-kp-emailProtection KeyPurposeIds in the same
+certificate.
+Note that the lint cannot distinguish cross-certificates from other
+intermediates.
+********************************************************************/
+
+package lints
+
+import (
+	"time"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+)
+
+func hasEKU(cert *x509.Certificate, eku x509.ExtKeyUsage) bool {
```

Comment:
I see a few other places that would benefit from this function. Can you promote it to the `util` package?

Reply:
Definitely, will move it there.